flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -281,6 +281,8 @@ public void notifyCheckpointAborted(long checkpointId, OperatorChain<?, ?> opera
 				}
 			}
 
+			channelStateWriter.abort(checkpointId, new CancellationException("checkpoint aborted via notification"), false);
```

Comment:
This is just an optimization (possible now with `abort` and `cleanup=false`).

Reply:
I like this optimization.
What happens if some other thread still enqueues data to this checkpoint? Or the writes dismissed?

Reply:
Yes, write requests will be ignored when dequeued by `ChannelStateWriteRequestExecutor`.